### ISSUE TITLE
spike lines do not appear when `spikedistance`

### ISSUE THREAD

**Issue by antoinerg** [plotly/plotly.js#4626]
At least in some traces: https://codepen.io/antoinerg/pen/BaNYJve

To be clear, this test will fail: https://github.com/plotly/plotly.js/commit/9d1325c75c1526caf174f0c4c93504c8c71c8e81

Affects traces:
- bar
- ohlc

**Comment by etpinard** [comment]
Duplicate of https://github.com/plotly/plotly.js/issues/3805 ?

**Comment by etpinard** [comment]
This problem probably affect all traces that use [`Fx.inbox`](https://github.com/plotly/plotly.js/blob/8f049fddbac0ca0382816984b8526857e9714fe6/src/components/fx/helpers.js#L82-L92) to determine which element getting hovered on.

**Comment by antoinerg** [comment]
Here's a potential fix for the current issue: https://github.com/plotly/plotly.js/compare/pr-unified-hoverlabel...fix-spikeline-infinity

I'm not sure yet whether it also fixes #3805. I will have to test.

**Comment by antoinerg** [comment]
It looks like it does fix issue #3805 as well:
Before: https://codepen.io/anon/pen/eoPXJE
After: https://codepen.io/antoinerg/pen/ExjQQYw

**Comment by etpinard** [comment]
Right, that makes things work for `spikedistance: -1`, but what about for arbitrary spike distances?

**Comment by etpinard** [comment]
(getting the `-1` case fixed would still be a major step forward though :smile: )

**Comment by antoinerg** [comment]
> Right, that makes things work for `spikedistance: -1`, but what about for arbitrary spike distances?

Those already work don't they?

**Comment by antoinerg** [comment]
Closing in favor of https://github.com/plotly/plotly.js/issues/3805
